From 8c1094a6596b8d05ce4072b1865520ff7f0accd2 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Fri, 18 Dec 2009 14:12:30 +0800
Subject: [PATCH 142/505] ENGR00119474 MX25: Fix camera overlay failure due to
 pan display offset

Camera capture test app will leave framebuffer pan display to none-zero
offset sometimes, this will affect overlay test. So make sure offset
to be zero before capture test is stopped.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 test/mxc_v4l2_test/csi_v4l2_capture.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/test/mxc_v4l2_test/csi_v4l2_capture.c b/test/mxc_v4l2_test/csi_v4l2_capture.c
index c0fb9fc..fedee1e 100644
--- a/test/mxc_v4l2_test/csi_v4l2_capture.c
+++ b/test/mxc_v4l2_test/csi_v4l2_capture.c
@@ -261,6 +261,13 @@ int v4l_capture_test(int fd_v4l)
 		gettimeofday(&tv2, NULL);
 	} while(tv2.tv_sec - tv1.tv_sec < g_timeout);
 
+	/* Make sure pan display offset is zero before capture is stopped */
+	if (var.yoffset) {
+		var.yoffset = 0;
+		if (ioctl(fd_fb, FBIOPAN_DISPLAY, &var) < 0)
+			printf("FBIOPAN_DISPLAY failed\n");
+	}
+
         if (stop_capturing(fd_v4l) < 0)
         {
                 printf("stop_capturing failed\n");
-- 
1.8.0

